### PR TITLE
Plugin Template Handler Support!

### PR DESCRIPTION
**Pull request by dangerousbeans**
Example template handler up here: [https://github.com/CollaboTree/volt_haml](https://github.com/CollaboTree/volt_haml)

### CODE REVIEW

File: spec/server/component_templates_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+else
+  require 'spec_helper'
+  require 'benchmark'
+  require 'volt/server/component_templates'
+end
+
+describe Volt::ComponentTemplates do
+  let(:ct){ Volt::ComponentTemplates.new('path/to/things', 'thing') }
+
+  describe "#available_formats" do
```
Comment (dfyx) [inline spec/server/component_templates_spec.rb]: This test fails on Travis with `NoMethodError: undefined method `[]' for Dir`

Comment (dangerousbeans) [inline spec/server/component_templates_spec.rb]: What does it mean? :C http://goo.gl/BMpXUr

**Comment by ryanstout** [comment]
Looks like specs are failing.  Merge with master, I did fix one issue in specs, though I think this is a different one.

**Comment by ryanstout** [comment]
Looks like there's some sort of order dependent failure here in the tests.  Let me know if you need help tracking it down.

**Comment by dangerousbeans** [comment]
WEIRD. I see this sometimes on my `rake` run, but not regularly. Any ideas?

**Comment by ryanstout** [comment]
Thanks.  If you want to add some docs on handlers, that would be great.  Also, feel free to add docs for using haml in the docs repo: https://github.com/voltrb/docs  Thanks!

**Pull request merged**